File: sys/s1.dart
Map global = new Map();

void log(a, [b='', c='', d='', e='', f='', g='', h='']) {
  print('${a} ${b} ${c} ${d} ${e} ${f} ${g} ${h}');
}

String str(o) {
    return o.toString();
}

List keys(o) {
    return o.keys.toList();
}

List values(o) {
    return o.values.toList();
}

List<int> range(int from, int to, [int step=1]) {
    List<int> list=[];
    int i=from;
    while (i<to) {
        list.add(i);
        i+=step;
    }
    return list;
}

List map(List a, dynamic Function(dynamic) f) {
    return a.map(f).toList();
}

List push(List list, dynamic a) {
    list.add(a);
    return list;
}

int len(List list) {
    return list.length;
}

File: sys/s1.py
# window.global = window

def log(*args):
    [print(arg, end='') for arg in args]
    print(' ')

def push(l, a):
    l.append(a)
    return l

"""
export function str(o) {
    if (o.__str) return o.__str()
    return o.toString()
}

export function keys(o) {
    if (o instanceof Map || o instanceof Set) return Array.from(o.keys())
    return Object.keys(o)
}

export function values(o) {
    if (o instanceof Map || o instanceof Set) return Array.from(o.keys())
    return Object.keys(o)
}

export function range(from, to, step=1) {
    let list=[], i=from;
    while (i<to) {
        list.push(i);
        i+=step;
    }
    return list;
}

export function re(exp, attr) {
    let r = new RegExp(exp, attr)
    r.lastIndex = 0
    return r
}

global.log = log
global.str = str
global.keys = keys
global.values = values
global.range = range
global.re = re
"""
